experimental/Rings/WeierstrassPreparationTheorem.jl

```diff
@@ -0,0 +1,214 @@
+##############################################################
+#
+#Conversion from Singular (weierstr.lib)
+#
+#############################################################
+
+"""
+`order_of_generality(f::MPolyElem)`
+
+compute for a given polynomial `f` an integer `d` such that `f` is `x_n`-general of order `d` with `x_n` the n-th variable of the polynomial ring of `f` 
+"""
```

Comment:
The docs need to be brushed up a bit, please have a look at
https://docs.oscar-system.org/dev/DeveloperDocumentation/documentation/
Also please wrap the lines to max 90-100 chars, I think that is the number we have in the style guide.